instrumentation/httptrace/httptrace_test.go

```diff
@@ -414,3 +414,291 @@ func TestStartRequestSpanMergedBaggage(t *testing.T) {
+	req.Header.Set("baggage", "user.id=,account.id=456,session.id=789,foo=bar")
+
+	// Start the request span, which will extract header baggage and merge it with the context's baggage.
+	span, _, _ := StartRequestSpan(req)
+	m := span.AsMap()
+	// if baggage header is malfromed, the entire header should be dropped
+	// therefore, there will be no baggage span tags
+	if _, ok := m["baggage.account.id"]; ok {
+		t.Errorf("baggage.account.id should not be included in span tags")
+	}
+	if _, ok := m["baggage.user.id"]; ok {
+		t.Errorf("baggage.user.id should not be included in span tags")
+	}
+	if _, ok := m["baggage.session.id"]; ok {
+		t.Errorf("baggage.session.id should not be included in span tags")
+	}
+	if _, ok := m["baggage.foo"]; ok {
+		t.Errorf("baggage.foo should not be included in span tags")
+	}
+	span.Finish()
+}
+
+func TestBaggageSpanTagsCaseSensitive(t *testing.T) {
+	tracer.Start()
+	defer tracer.Stop()
+
+	// Create an HTTP request with that context.
+	req := httptest.NewRequest(http.MethodGet, "/somePath", nil).WithContext(context.Background())
+
+	// Set the baggage header with additional baggage items.
+	req.Header.Set("baggage", "user.id=doggo,ACCOUNT.id=456,seSsIon.id=789")
+
+	// Start the request span, which will extract header baggage and merge it with the context's baggage.
+	span, _, _ := StartRequestSpan(req)
+	m := span.AsMap()
+	// no values should be present in span tags
+	if _, ok := m["baggage.account.id"]; ok {
+		t.Errorf("baggage.account.id should not be included in span tags")
+	}
+	if _, ok := m["baggage.session.id"]; ok {
+		t.Errorf("baggage.session.id should not be included in span tags")
+	}
+	if v, ok := m["baggage.user.id"]; ok {
+		assert.Equal(t, "doggo", v, "should contain user.id value")
+	} else {
+		t.Errorf("baggage.user.id not found in span")
+	}
+	span.Finish()
+}
+
+func TestBaggageSpanTagsExtractionOnly(t *testing.T) {
+	tracer.Start()
+	defer tracer.Stop()
+	ctx := baggage.Set(context.Background(), "user.id", "alice")
+	ctx = baggage.Set(ctx, "account.id", "testaccount")
+	ctx = baggage.Set(ctx, "region", "us-east-1")
+
+	// Create an HTTP request with context.
+	req := httptest.NewRequest(http.MethodGet, "/somePath", nil).WithContext(ctx)
+
+	// Set the baggage header with additional baggage items.
+	req.Header.Set("baggage", "user.id=john,session.id=testingsession")
+
+	// Start the request span, which will extract header baggage and merge it with the context's baggage.
+	span, _, _ := StartRequestSpan(req)
+	m := span.AsMap()
+	// baggage span tagging happens only at extraction
+	if _, ok := m["baggage.account.id"]; ok {
+		t.Errorf("baggage.account.id should not be included in span tags")
+	}
+	if _, ok := m["baggage.region"]; ok {
+		t.Errorf("baggage.region should not be included in span tags")
+	}
+	if v, ok := m["baggage.user.id"]; ok {
+		assert.Equal(t, "john", v, "should contain user.id value")
+	} else {
+		t.Errorf("baggage.user.id not found in span")
+	}
```

Comment:
```suggestion
	v, ok := m["baggage.user.id"]
	assert.True(t, "baggage.user.id not found in span")
	assert.Equal(t, "john", v, "should contain user.id value")
```

How about using something like above? Easier on the eyes.

https://pkg.go.dev/github.com/stretchr/testify/assert#True